remotes/docker/resolver.go

```diff
@@ -354,6 +354,11 @@ func (r *dockerBase) retryRequest(ctx context.Context, req *http.Request, respon
 			req.Method = http.MethodGet
 			return copyRequest(req)
 		}
+	} else if last.StatusCode == http.StatusForbidden {
+		if r.username != "" && r.secret != "" {
+			r.useBasic = true
```

Comment:
Re-trying with credentials after a 403 is not an expected flow. Also `WWW-Authenticate` is not an expected header for a 403 so I don't think we can rely on that. Do we need to reach out to the GCR team to figure this one out, is their expectation that credentials will always be passed into the initial request? Curious if only the `/v2/` endpoint is returning the 401 while the rest are returning a 403. Either way we need more information, and if we were to implement a work around here, it need to be accompanied with a thorough explanation for its inclusion.